rest-server/src/templates/dockerContainerScript.mustache

```diff
@@ -46,20 +50,27 @@ export PAI_KILL_ALL_ON_COMPLETED_TASK_NUM={{{ jobData.killAllOnCompletedTaskNumb
 
 export PAI_CURRENT_CONTAINER_PORT=$(( (RANDOM % 55535) + 10001 ))
 hdfs dfs -touchz $HDFS_LAUNCHER_PREFIX/$PAI_USERNAME/$PAI_JOB_NAME/tmp/$PAI_TASK_ROLE_NO-$PAI_TASK_INDEX-$PAI_CURRENT_CONTAINER_IP-$PAI_CURRENT_CONTAINER_PORT || exit 1
-while [ `hdfs dfs -ls $HDFS_LAUNCHER_PREFIX/$PAI_USERNAME/$PAI_JOB_NAME/tmp/ | wc -l` -lt  $PAI_TASKS_NUM ]; do
+while [ `hdfs dfs -ls $HDFS_LAUNCHER_PREFIX/$PAI_USERNAME/$PAI_JOB_NAME/tmp/ | grep "/$PAI_JOB_NAME/tmp" | wc -l` -lt  $PAI_TASKS_NUM ]; do
+  printf "%s\n%s\n\n" "[INFO]" "Waiting for other containers ..."
   sleep 10
 done
 hdfs dfs -ls $HDFS_LAUNCHER_PREFIX/$PAI_USERNAME/$PAI_JOB_NAME/tmp/ \
   | grep "/$PAI_JOB_NAME/tmp" \
   | grep -oE "[^/]+$" \
   | sort -n -k 2 -t"-" \
   > ContainerList
+if [ "$(cat ContainerList | wc -l)" -ne $PAI_TASKS_NUM ]; then
```

Comment:
Is this line necessary in production code? Or is it just for tenporary debugging?

Reply:
yes. it exits

Reply:
we assume gang scheduling. if the code hit here, something critically wrong happen. we have to exit. In the future, we might support elastic computing, which is another story.

---
In reply to: [161922808](https://github.com/Microsoft/pai/pull/235#discussion_r161922808) [](ancestors = 161922808)